File: api/model/sandbox.go
package model

import (
	"time"
)

type Sandbox struct {
	Id           uint      `json:"id"`
	Name         string    `json:"name"`
	Language     string    `json:"language"`
	Code         string    `json:"code"`
	Status       string    `json:"status"`
	StdOut       string    `json:"stdout"`
	StdErr       string    `json:"stderr"`
	ExecDuration int       `json:"execDuration" gorm:"type:integer; null"`
	ExecMemUse   int       `json:"execMemUse" gorm:"type:integer; null"`
	UserRefer    uint      `json:"userRefer"`
	CreatedAt    time.Time `json:"createdAt"`
	UpdatedAt    time.Time `json:"updatedAt"`
}

File: client/models/sandbox.ts
import { z } from 'zod';

export const ZSandbox = z.object({
  id: z.number(),
  name: z.string(),
  language: z.string(),
  code: z.string(),
  status: z.string(),
  stdout: z.string(),
  stderr: z.string(),
  execDuration: z.number(),
  execMemUse: z.number(),
  createdAt: z.string(),
  updatedAt: z.string(),
  userRefer: z.number(),
});
